core/trino-main/src/main/java/io/trino/sql/planner/LocalExecutionPlanner.java

```diff
@@ -581,7 +581,8 @@ public LocalExecutionPlan plan(
         PartitionFunction partitionFunction;
         Optional<SkewedPartitionRebalancer> skewedPartitionRebalancer = Optional.empty();
         int taskCount = getTaskCount(partitioningScheme);
-        if (checkCanScalePartitionsRemotely(taskContext.getSession(), taskCount, partitioningScheme.getPartitioning().getHandle(), nodePartitioningManager)) {
+        if (SystemSessionProperties.getRetryPolicy(taskContext.getSession()) != RetryPolicy.TASK
```

Comment:
nit: what happens if I change policy from `TASK` to none via Session property?

Reply:
So NONE is no retries - and the we want to use skewedPartitionRebalancer

Reply:
I would guess using none on fte cluster has perf implications